src/language-js/print/ternary-old.js

```diff
@@ -274,16 +274,42 @@ function printTernaryOld(path, options, print) {
         : wrap(print(alternateNodePropertyName)),
     );
   } else {
+    /*
+    This does not mean to indent, but make the doc aligned with the first character after `? ` or `: `,
+    so we use `2` instead of `options.tabWidth` here.
+    
+    ```js
+    test
+     ? {
+         consequent
+       }
+     : alternate
+    ```
+
+    instead of
+    
+    ```js
+    test
+     ? {
+       consequent
+     }
+     : alternate
+    ```
+    */
+    const printBranch = (nodePropertyName) =>
+      options.useTabs
+        ? indent(print(nodePropertyName))
+        : align(2, print(nodePropertyName));
     // normal mode
     const part = [
       line,
       "? ",
       consequentNode.type === node.type ? ifBreak("", "(") : "",
-      align(2, print(consequentNodePropertyName)),
+      printBranch(consequentNodePropertyName),
       consequentNode.type === node.type ? ifBreak("", ")") : "",
       line,
       ": ",
-      align(2, print(alternateNodePropertyName)),
+      printBranch(alternateNodePropertyName),
```

Comment:
I think we should use `?\t` instead of `? `.

When user change `tabWidth`, this is not aligned.
```
test
	? {
			consequent
		}
	: alternate
```

But this still aligned.

```
test
	?	{
			consequent
		}
	: alternate
```

Reply:
ternary.js works this way.  

https://github.com/prettier/prettier/blob/093745f0ec429d3db47c1edd823357e0ef24e226/src/language-js/print/ternary.js#L381

Reply:
I just checked how I wrote this when use tabs. But I found that I use `?<SPACE>`, not `?<TAB>`.

I understand ternary.js checked [isBigTabs](https://github.com/prettier/prettier/blob/093745f0ec429d3db47c1edd823357e0ef24e226/src/language-js/print/ternary.js#L182C9-L182C18), but I don't thinks when using `tab`s, the `tabWidth` matters, since it will be set by editor not by us.

@rattrayalex @sosukesuzuki Any comment?

Reply:
I think I misunderstood how this works in the new ternary format, the solution looks perfect there.

Reply:
The diff is huge, but with `?\t` it looks much better https://github.com/prettier/prettier/pull/15662/commits/2a37d8c219fb78f83c80a05afdad7b550d5cb216

Reply:
I'm not sure you *always* want to use a tab after `?` or `:`.

In particular, this (from near the end of the diff) is definitely wrong:
`const obj0 = conditionIsTruthy ?	shortThing :	otherShortThing;`

Because `const obj0 = conditionIsTruthy ?` does not have a length of 1, `shortThing` is going to jump around depending on the tab width, and so is `otherShortThing`. Tabs *after* text can only consistently align things if the text is *always* short enough to fit inside a single tab - which is only true for a single character since you could have a tab width of 1.

Reply:
We probably don't want to cause a huge diff if we are going to change to the new "curious ternaries" format soon anyway. So we may want to wait until that is decided.

Otherwise I have nothing to add to the above discussion, sounds like you have it figured out 👍

Reply:
I think you should be a little more careful about this. The scope of influence is too large.

Reply:
I prefer only resolve the original issue ( #15655 ) in this PR and discuss about tabs after `?` as a separate issue.